README.md

```diff
@@ -98,9 +98,6 @@ coverage, expressiveness of executable protocol traces and stable and extensible
 * clang
 * graphviz
 
-OpenSSL 1.0:
-* makedepend from `xutils-dev package
```

Comment:
Nope. I removed the dependency in #300 (on which this PR is stacked).

I had several instances of makedepend not finding the correct dependencies. In the best case, the build failed but sometimes it caused subtle problems with part of the system OpenSSL being included in our build. Since makedepend was a pain to setup on some platforms, I chose to patch it away definitely.

You can see `puts/vendor/scripts/openssl/CMakeLists.txt` for the changes: I patch the OpenSSL build system and use the compiler to extract dependencies instead. This is the way it is done in more recent releases of OpenSSL and any modern compiler we plan to support should be able to do that.